lib/query-interface/knex.js

```diff
@@ -0,0 +1,41 @@
+const QueryInterface = require('./base');
+
+class KnexQueryInterface extends QueryInterface {
+  async query(statement, { transaction, replacements, firstResult }) {
+    const result = await this.queryInterface.raw(statement, replacements).transacting(transaction);
+    return firstResult ? result.rows[0] : result.rows;
+  }
+
+  async createTransaction() {
+    return this.queryInterface.transaction();
+  }
+
+  getDialectName() {
+    const { client } = this.queryInterface.connection().client.config;
+    switch (client) {
+      case 'pg':
+        return 'postgres';
+      default:
+        return client;
+    }
+  }
+
+  parseDatabaseError(error) {
+    if (
+      error.message && error.message.indexOf('could not obtain lock') > -1
```

Comment:
so you can't catch an error by it's class in node? or is knex throwing a generic `Error` here?

Reply:
knex is throwing a generic `Error` yea